src/controller/python/README.md

```diff
@@ -1,244 +1,255 @@
+   Done
+
+   sudo ot-ctl dataset extpanid
+   4fe76e9a8b5edaf5
+   Done
+   ```
+
+   Matter specifiction does not define how the Thread or Wi-Fi credentials are obtained by Controller. For example, for Thread, instead of fetching datasets directly from the Thread Border Router, you may also use different out-of-band method.
+
+2. Inject the previously obtained Active Operational Dataset as hex-encoded value using ZCL Network Commissioning cluster.
+
+   > Each ZCL command has a following format:
+      `zcl <Cluster> <Command> <Node Id> <Endpoint Id> <Group Id> [arguments]`
+
+   ```
+   chip-device-ctrl > zcl NetworkCommissioning AddThreadNetwork 1234 0 0 operationalDataset=hex:0e080000000000010000000300001335060004001fffe002084fe76e9a8b5edaf50708fde46f999f0698e20510d47f5027a414ffeebaefa92285cc84fa030f4f70656e5468726561642d653439630102e49c0410b92f8c7fbb4f9f3e08492ee3915fbd2f0c0402a0fff8 breadcrumb=0 timeoutMs=3000
+   ```
+
+3. Enable Thread interface in the device by executing the following command with `networkID` equal to Extended PAN Id of the Thread network.
 
-## IP Secure Session Establishment
+   ```
+   chip-device-ctrl > zcl NetworkCommissioning EnableNetwork 1234 0 0 networkID=hex:4fe76e9a8b5edaf5 breadcrumb=0 timeoutMs=3000
+   ```
 
-1. Run CHIP Device Controller
+#### Commissioning of Wi-Fi device
+
+1. Assuming your Wi-Fi SSID is *TESTSSID*, and your Wi-Fi password is *P455W4RD*, inject the credentials to the device by excuting the following command.
+
+   ```
+   chip-device-ctrl > zcl NetworkCommissioning AddWiFiNetwork 1234 0 0 ssid=str:TESTSSID credentials=str:P455W4RD breadcrumb=0 timeoutMs=1000
+   ```
+
+2. Enable Wi-Fi interface in the device by executing the following command:
+
+   ```
+   chip-device-ctrl > zcl NetworkCommissioning EnableNetwork 1234 0 0 networkID=str:TESTSSID breadcrumb=0 timeoutMs=1000
+   ```
+
+### 6. Close Bluetooth LE connection.
+
+After Matter accessory was provisioned with Thread and/or Wi-Fi credentials, the commissioning process is finished. Python CHIP controller will now use only IPv6 traffic to reach the device, so Bluetooth LE connection can be closed.
 
 ```
-chip-device-ctrl
+chip-device-ctrl > close-ble
 ```
 
-> Note: SUDO is not required when use IP to connect device.
+### 7. Discover IPv6 address of the Matter accessory.
+
+The Matter controller needs to discover IPv6 address of the node that it previously commissioned. For Thread, the Matter accessory uses SRP (Service Registration Protocol) to register its presence on the Thread Border Router’s SRP Server, for Wi-Fi or Ethernet devices, the mDNS (Multicast Domain Name System) protocol is used instead.
 
-2. Connect to device using setup pin code
+Run the following command, assuming that:
+ - The Fabric ID of the device is 5544332211
+ - The Node ID is 1234
 
 ```
-chip-device-ctrl > connect -ip <Device IP Address> 20202021
+chip-device-ctrl > resolve 5544332211 1234
```

Comment:
Can the Node ID be ommitted here as well? Or is this the place where we use the Node ID that we noted down? If so, please mention it in the introductory line: "(use the Node ID you noted down when you established the secure connection over Bluetooth LE)".